app/src/crypto.c

```diff
@@ -62,11 +87,9 @@ uint16_t crypto_fillAddress_secp256k1(uint8_t *buffer, uint16_t buffer_len) {
 
     crypto_extractPublicKeyHash(address_temp.hash_ripe, CX_RIPEMD160_SIZE);
 
-    uint8_t version = COIN_VERSION_MAINNET_SINGLESIG;
-    if (isTestnet()) {
-        version = COIN_VERSION_MAINNET_SINGLESIG;
-    }
     size_t outLen = sizeof_field(answer_t, address);
+    if ( !is_valid_network_version(version) )
+        version = COIN_VERSION_MAINNET_SINGLESIG;
```

Comment:
Would it be better/safer to throw an error here, or just accept the unknown version byte, than to default to mainnet single sig?

Reply:
yes, I agree

Reply:
yes,  I agree as well. Will update this.